src/core_plugins/vega/public/vega_visualization.js

```diff
@@ -0,0 +1,67 @@
+import { Notifier } from 'ui/notify';
+import { VegaView } from './vega_view/vega_view';
+
+export function VegaVisualizationProvider(vegaConfig, serviceSettings) {
+
+  const notify = new Notifier({ location: 'Vega' });
+
+  return class VegaVisualization {
+    constructor(el, vis) {
+      this._el = el;
+      this._vis = vis;
+    }
+
+    /**
+     *
+     * @param {VegaParser} visData
+     * @param {*} status
+     * @returns {Promise<void>}
+     */
+    async render(visData, status) {
+      if (!visData && !this._vegaView) {
+        notify.warning('Unable to render without data');
+        return;
+      }
+
+      try {
+
+        await this._render(visData, status);
+
+      } catch (error) {
+        if (this._vegaView) {
+          this._vegaView.onError(error);
+        } else {
+          notify.error(error);
+        }
+      }
+    }
+
+    async _render(vegaParser, status) {
+      if (vegaParser && (status.data || !this._vegaView)) {
+
+        // New data received, rebuild the graph
+        if (this._vegaView) {
+          await this._vegaView.destroy();
+          this._vegaView = null;
+        }
+
+        if (vegaParser.useMap) {
+          throw new Error('Map mode is not yet supported in Kibana Core. You must use Kibana Vega plugin');
```

Comment:
I think the error message is wrong? It should be "You must use the Kibana **map** plugin" (and I would replace plugin by "visualization").

Reply:
the Vega plugin supports the map mode, but in the core we only support regular vega, without the map capability.  Users who want to place visualization on top of a dynamic (pannable/zoomable) map need to install kibana vega plugin and do the whole visualization there.

Reply:
Maybe we could then write "You must use either the Kibana Vega plugin or Kibana map visualizations.". Also I am not so sure, what's our plan about having core plugins advertise "3rd party" plugins (@thomasneirynck ?)

Also does it work installing the plugin beside? Do they use different visualization type names, or would they clash (no idea how vis behaves, if two code vis classes sign up for the same id)?

Reply:
@timroes they do work side by side (i have been testing that scenario too). The only people who would ever see this message are the ones who try to migrate from the plugin to the core, so this is not exactly an advertisement :). The map mode is not documented in Kibana docs, only in the plugin page. If someone tries to enable it, they must have come from the plugin.  I do hope to migrate tile-based map support from the plugin as well, just not yet.  Also, the plugin supports everything that core supports - so it is always up to the user - use plugin or the (more limited) core.
As for the text  -- I'm just not sure about the "either" in this context -- this message is basically telling those who migrate that this specific feature is not yet available in the core, and they should continue using the plugin until it does make it into core. They can use plugin for everything, or just for the map-based vis, or any other mix-and-match approach.